src/components/CIPipelineN/ConditionContainer.tsx

```diff
@@ -0,0 +1,318 @@
+        selectedTaskIndex,
+        activeStageName,
+    }: {
+        formData: FormType
+        setFormData: React.Dispatch<React.SetStateAction<FormType>>
+        selectedTaskIndex: number
+        activeStageName: string
+    } = useContext(ciPipelineContext)
+    const operatorOptions: OptionType[] = [
+        { value: '==', description: 'equal to' },
+        { value: '!=', description: 'not equal to' },
+        { value: '<', description: 'less than' },
+        { value: '>', description: 'greater than' },
+        { value: '<=', description: 'less than or equal to' },
+        { value: '>=', description: 'greater than or equal to' },
+    ].map((operator) => ({ label: operator.value, value: operator.value, description: operator.description }))
+    const [collapsedSection, setCollapsedSection] = useState<boolean>(true)
+    const [selectedOperator, setSelectedOperator] = useState<OptionType>(operatorOptions[0])
+    const [conditionType, setConditionType] = useState<ConditionType>(
+        type === ConditionContainerType.PASS_FAILURE ? ConditionType.SUCCESS : ConditionType.TRIGGER,
+    )
+
+    const currentStepTypeVariable =
+        formData[activeStageName].steps[selectedTaskIndex].stepType === PluginType.INLINE
+            ? 'inlineStepDetail'
+            : 'pluginRefStepDetail'
+
+    useEffect(() => {
+        setCollapsedSection(true) // collapse all the conditions when user go from prebuild to post build
+    }, [activeStageName])
+
+    const addCondition = (): void => {
+        const _formData = { ...formData }
+        let id = 0
+        let conditionTypeToRemove
+        if (type === ConditionContainerType.PASS_FAILURE) {
+            conditionTypeToRemove = conditionType === ConditionType.SUCCESS ? ConditionType.FAIL : ConditionType.SUCCESS
+        } else {
+            conditionTypeToRemove = conditionType === ConditionType.TRIGGER ? ConditionType.SKIP : ConditionType.TRIGGER
+        }
+        if (!_formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable].conditionDetails) {
+            _formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable].conditionDetails = []
+        }
+        for (
+            var i = 0;
+            i < _formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable].conditionDetails.length;
+            i++
+        ) {
+            if (
+                _formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable].conditionDetails[i]
+                    .conditionType === conditionTypeToRemove
+            ) {
+                _formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable].conditionDetails.splice(
+                    i,
+                    1,
+                )
+                i--
+            } else {
+                id =
+                    id <
+                    _formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable].conditionDetails[i].id
```

Comment:
try to move `_formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable].conditionDetails` to some variable above